minio/scripts/file-operations.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+aws_cli() {
+  aws --endpoint-url http://minio:9000 $@
+}
+
+set -e
+
+# Executable that waits for a port at an address to be open.
+# For further details go to: https://github.com/ufoscout/docker-compose-wait
+/wait
+
+echo "Create private bucket (private by default)"
+aws_cli s3 mb s3://private-bucket
+
+echo "Upload file to private bucket with custom meta data"
+aws_cli s3 cp ./assets/file.txt s3://private-bucket --content-type text/plain --metadata CustomMetaKey=CustomMetaValue
```

Comment:
it's good to be able to set metadata, but the key names have a [special meaning](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html#object-metadata) that can lead a file to be encrypted or to produce a redirect to something else. What we want for arbitrary key-value pairs is `aws s3api put-object-tagging`, or an equivalent in `aws s3 cp` if it has an option like that (couldn't find it).

Reply:
Looks like minio might not support this api yet: `An error occurred (NotImplemented) when calling the PutObjectTagging operation: A header you provided implies functionality that is not implemented`

Reply:
So we have to keep metadata as a fallback and avoid using keys that are reserved for AWS e.g. `x-amz` ones or `Content-Length`.

Reply:
adding to the conversation for record https://github.com/minio/minio/blob/master/cmd/generic-handlers.go#L473

Reply:
updating to use `x-amz-meta-mytagname`

Reply:
updated.